lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -159,9 +171,7 @@ def _on_relation_event(self, event):
 
         departed_relation_id = None
         if isinstance(event, RelationBrokenEvent):
-            departed_relation_id = self.charm.proceed_on_broken_event(event)
-            if not departed_relation_id:
-                return
+            departed_relation_id = event.relation.id
```

Comment:
Looks like here if the event is NOT  `RelationBrokenEvent`  then `departed_relation_id` will be None, so maybe in that case we need to return from function earlier?

Reply:
> Looks like here if the event is NOT RelationBrokenEvent then departed_relation_id will be None

That is correct. I am not 100% sure what you mean by `in that case we need to return from function earlier`, we should **not** return if `departed_relation_id` is `None`

In this function we pass `departed_relation_id` to subsequent functions and those functions handle the cases in which `departed_relation_id` is `None`, namely:
```
            self.add_shards(departed_relation_id)
            self.remove_shards(departed_relation_id)
 ```
 
 does that answer your question? Or am I misunderstanding?

Reply:
Why would we want to do his call ?

```
self.add_shards(None)
self.remove_shards(None)
```

Reply:
Please read the functions `add_shards` and `remove_shards` 

ideally from those function signatures it should be obvious why we would want to do this. If it isn't obvious then there is a big problem in the readability of this file and that needs to be updated ASAP